ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Tables/Utils.hs

```diff
@@ -0,0 +1,194 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+-- | A collection of useful combinators to shorten the code in other places.
+--
+-- FIXME (jasagredo): I don't like this module, it is super contraintuitive and
+-- you never know if it is going to keep the first or the second ledger state.
```

Comment:
If we are explicit about this in the haddock comments, at least the downstream code will be aware, right?

Reply:
Good question. I was kind of leaving this for the last moment :smile:

Reply:
We should at least remove the `FIXME (jasagredo)` and replace by a `TODO` since this does not seem critical.

Reply:
Done